### ISSUE TITLE
GDPR compliance

### ISSUE THREAD

**Issue by mtrezza** [parse-community/parse-server#5378]
**TLDR**: This feature request describes data access and modification logging which is a legal requirement in some jurisdictions.

---
**Is your feature request related to a problem? Please describe.**
Parse Server offers data access and manipulation via its Parse Dashboard. For the dashboard to be usable in a business (or even "hobby developer") environment, Parse Server needs to comply with GDPR if it handles data of "EU users"*.

The dashboard is the main (and only?) tool of Parse Server for quick and easy manual data view and manipulation. It would be beneficial for many users if the dashboard (in connection with parse server) complied with GDPR.

**Describe the solution you'd like**
A mechanism that logs:

- User login
- Data view
- Data manipulation (entry / modification)
- Schema manipulation
- ACL manipulation
- Push sending

The easiest way would probably be to add a logging mechanism to the Parse Dashboard alone, without any modification of Parse Server, if that's possible. It should store the logs in a separate file, so these logs files can be easily dealt with according to archiving requirement of GDPR (audit-proof, etc).

**Describe alternatives you've considered**
Not making the dashboard GRPD compliant would render it legally unusable for aforementioned data.
Alternatives:
- Data manipulation directly in the DB; that shifts the GDPR compliance requirement to the DB interface.
- Create a separate app with user management and logging, essentially replicating functions of the parse dashboard.

**Additional context**
* [GDPR](https://en.wikipedia.org/wiki/General_Data_Protection_Regulation) compliance requires the logging of data access and data modification (among many other things) for data of EU residents (regardless of citizenship) and EU citizens (regardless of residency). GDPR is enforceable since 25 May 2018.

**Comment by acinader** [comment]
I assume that doing this at the data level (i.e. mongo logging) would not be adequate cause we don't have the user associated with the changes.

Adding to the dashboard seems reasonable to me.  Would you be willing to open a pr to get it started?

**Comment by mtrezza** [comment]
Logging on DB level was just meant as alternative if someone decides to **not** use Parse Dashboard and wants to be GDPR compliant.

Without committing to a PR at this point, I would like to start a conceptual discussion about how / where to implement it. So we can estimate the effort and someone can pick it up.

**Comment by mrmarcsmith** [comment]
Even for apps that don’t require GDPR I think this is a awesome idea. I don’t have a ton of time to help with a PR but would love to weigh in on the conceptual. I would much rather see the “server centric” solution than the parse dashboard one aswell. “In theory” those with dashboard access could potentially retrieve the master key and cercumvent the dashboard logging all together. Where as, when it’s baked into parse server there isn’t any way around the logging.

**Comment by mtrezza** [comment]
don't close, open issue

**Comment by mtrezza** [comment]
don't close, still relevant

**Comment by TomWFox** [comment]
It won’t be closed now I’ve added the ‘help wanted’ label 🙂

**Comment by mtrezza** [comment]
Stumbled across Parse Auditor, which addresses this issue. Not sure yet whether it is fully GDRP compliant.

https://github.com/Blackburn-Labs/parse-auditor

**Comment by mtrezza** [comment]
Related: https://github.com/netreconlab/parse-hipaa

**Comment by dblythy** [comment]
Is this in progress at all?

**Comment by RahulLanjewar93** [comment]
> Stumbled across Parse Auditor, which addresses this issue. Not sure yet whether it is fully GDRP compliant.
> 
> https://github.com/Blackburn-Labs/parse-auditor

This package overwrites the triggers, which is not feasible. So I created another package that pretty much does the same. 
It's not GDPR Compliant but if someone wants to use it here's the link 
https://github.com/RahulLanjewar93/parse-audit-log

Suggestions are appreciated thanks!
